resources/views/profile.blade.php

```diff
@@ -33,25 +30,46 @@
                         @if(session('success'))
                             <p class="alert {{ Session::get('alert-class', 'alert-success') }}">{{ Session::get('success') }}</p>
                         @endif
+                        @if($permission==config('constants.FRIENDS') || $user->username==Auth::user()->username)
                         <table class="table table-striped table-bordered table-hover text-center control-label">
                             <tr class="active">
                                 <td>First Name: </td>
-                                <td>{{ Auth::user()->firstname }}</td>
+                                <td>{{ $user->firstname }}</td>
                             </tr>
                             <tr class="active">
                                 <td>Last Name: </td>
-                                <td>{{ Auth::user()->lastname }}</td>
+                                <td>{{ $user->lastname }}</td>
                             </tr>
                             <tr class="active">
                                 <td>Email: </td>
-                                <td>{{ Auth::user()->email }}</td>
+                                <td>{{ $user->email }}</td>
                             </tr>
                             <tr class="active">
                                 <td>Biography: </td>
-                                <td>{{ Auth::user()->bio }}</td>
+                                <td>{{ $user->bio }}</td>
                             </tr>
                         </table>
+                          @if($permission==config('constants.FRIENDS'))
+                          <h3>You and {{$user->username}} are friends! Buy lots of things together!</h3>
+                          @endif
+                        @endif
+                        @if($permission==config('constants.NOT_FRIENDS'))
+                        <form method="POST" action="/search">
+                          <input type="hidden" name="_token" value="{{ csrf_token() }}">
+                          <span class="pull-right">
```

Comment:
this button needs to be hidden if you're looking at your own profile

![image](https://cloud.githubusercontent.com/assets/15661248/24888911/9960c460-1e1b-11e7-9af3-0a12be873a42.png)